OpenRA.Mods.Common/Traits/Harvester.cs

```diff
@@ -118,20 +119,24 @@ public Harvester(Actor self, HarvesterInfo info)
 
 		void INotifyCreated.Created(Actor self)
 		{
+			notify = self.TraitsImplementing<INotifyHarvesterAction>().ToArray();
+
 			if (Info.SearchOnCreation)
 				self.QueueActivity(new FindResources(self));
 		}
 
 		void INotifyBuildComplete.BuildingComplete(Actor self)
 		{
+			notify = self.TraitsImplementing<INotifyHarvesterAction>().ToArray();
```

Comment:
> Ok, then how about we put `notify` *only* in Created, and the `QueueActivity` *only* in BuildingComplete?

The former is fine, but the latter won't work because harvs spawned from refinery don't trigger BuildingComplete.